app/assets/src/components/visualizations/dendrogram/Dendogram.js

```diff
@@ -147,6 +150,113 @@ export default class PhyloTree {
     });
   }
 
+  updateColors() {
+    // Color clusters of the nodes based on the color group attribute.
+    // Color a node if all its children have the same color.
+    if (!this.options.colorGroupAttribute) {
+      return;
+    }
+
+    let attrName = this.options.colorGroupAttribute;
+
+    // Set up all attribute values. Colors end up looking like:
+    // Uncolored (grey) | Real values.. | Absent attribute color (e.g. for External Sources)
+    let allVals = new Set();
+    this.root.leaves().forEach(n => {
+      if (n.data && n.data[attrName]) {
+        allVals.add(n.data[attrName]);
+      }
+    });
+    allVals = Array.from(allVals);
+    allVals = ["Uncolored"]
+      .concat(allVals)
+      .concat(this.options.colorGroupAbsentName);
+    this.allColorAttributeValues = allVals;
+
+    // Set up colors array
+    this.colors = Colormap.getNScale(this.options.colormapName, allVals.length);
+    this.colors = ["#cccccc"].concat(this.colors);
```

Comment:
Can you move the color to an option?
I just do not link having magic values in the code.

Reply:
👍